### PR TITLE
Fix regression issue that skipped every other frame

### PR DESCRIPTION
**Pull request by jafyvilla**
Recovered the --maxFrames option, which can be used together with
--frames to have animation loops

### CODE REVIEW

File: livre/Eq/Window.cpp [hunk 1]
```diff
@@ -91,22 +91,23 @@ class Window::Impl
...
         if( pipe->getFrameData()->getAppParameters()->animationEnabled )
-            renderStatus.setFrameID( startFrame + frameNumber - 1 );
-        else
-            renderStatus.setFrameID( startFrame );
-
-        commit();
+        {
+            const Vector2ui& animationFrames = pipe->getFrameData()->getAppParameters()->frames;
+            const uint32_t duration = animationFrames.y() - animationFrames.x();
+            renderStatus.setFrameID( animationFrames.x() + ( frameNumber % duration ));
```
Comment (bilgili) [inline livre/Eq/Window.cpp]: So loop mode is implicit ?

Comment (jafyvilla) [inline livre/Eq/Window.cpp]: It is if the number of maxFrames is larger than the frame range for the animation.

File: livre/Eq/Window.cpp [hunk 2]
```diff
@@ -91,22 +91,23 @@ class Window::Impl
...
+        }
```
Comment (eile) [inline livre/Eq/Window.cpp]: Why is this in a subscope?

File: livre/Lib/Configuration/ApplicationParameters.cpp [hunk 3]
```diff
@@ -114,8 +120,18 @@ void ApplicationParameters::initialize_()
...
 }
+
+namespace lunchbox
+{
+template<> inline void byteswap( vmml::Vector2ui& value )
+{
+    byteswap( value.x( ));
+    byteswap( value.y( ));
+}
+}
```
Comment (eile) [inline livre/Lib/Configuration/ApplicationParameters.cpp]: Should be in ../Equalizer//eq/fabric/vmmlib.h

File: livre/Eq/Window.cpp (hunk 2 above)
Comment (jafyvilla) [inline livre/Eq/Window.cpp]: Because this only makes sense when animation mode is enabled (line 105)

**Comment by bilgili** [comment]
retest this please

File: livre/Lib/Configuration/ApplicationParameters.cpp (hunk 3 above)
Comment (jafyvilla) [inline livre/Lib/Configuration/ApplicationParameters.cpp]: https://github.com/Eyescale/Equalizer/pull/458

File: livre/Eq/Window.cpp (hunk 2 above)
Comment (eile) [inline livre/Eq/Window.cpp]: Ah, missed the if()

**Comment by eile** [comment]
+1

**Comment by bilgili** [comment]
retest this please

**Comment by jafyvilla** [comment]
retest this please

**Pull request merged**